integration-cli/docker_cli_run_test.go

```diff
@@ -4415,3 +4415,17 @@ func (s *DockerSuite) TestRunHostnameInHostMode(c *check.C) {
 	out, _ := dockerCmd(c, "run", "--net=host", "--hostname=foobar", "busybox", "sh", "-c", `echo $HOSTNAME && hostname`)
 	c.Assert(strings.TrimSpace(out), checker.Equals, expectedOutput)
 }
+
+func (s *DockerDaemonSuite) TestRunWithProxyEnv(c *check.C) {
+	expectedOutput := "HTTP_PROXY=127.0.0.1:3001"
+	s.d.StartWithBusybox(c, "--debug", "--proxy-env", expectedOutput)
+	out, _ := dockerCmd(c, "run", "busybox", "env")
```

Comment:
😚 that. is. amazing. it's bothered me since the dawn of this PR and I hadn't even noticed this hiding in plain sight. THANK YOU!